common_rest/src/test/java/uk/gov/dhsc/htbhf/errorhandler/ErrorHandlerTest.java

```diff
@@ -0,0 +1,148 @@
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.mock.http.MockHttpInputMessage;
+import org.springframework.validation.BindException;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.context.request.WebRequest;
+import uk.gov.dhsc.htbhf.requestcontext.RequestContext;
+
+import java.nio.charset.Charset;
+import java.time.LocalDate;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static uk.gov.dhsc.htbhf.errorhandler.ErrorHandler.UNREADABLE_ERROR_MESSAGE;
+import static uk.gov.dhsc.htbhf.errorhandler.ErrorHandler.VALIDATION_ERROR_MESSAGE;
+
+@ExtendWith(MockitoExtension.class)
+class ErrorHandlerTest {
+
+    private static final String REQUEST_ID = "myRequestId";
+    private static final String ERROR_MESSAGE_1 = "error1";
+    private static final String ERROR_MESSAGE_2 = "error2";
+
+    @Mock
+    BindingResult bindingResult;
+    @Mock
+    HttpHeaders httpHeaders;
+    @Mock
+    WebRequest webRequest;
+    @Mock
+    RequestContext requestContext;
+
+    @InjectMocks
+    ErrorHandler handler;
+
+    @Test
+    void shouldExtractFieldErrorsFromBindException() {
+        // Given
+        FieldError fieldError1 = new FieldError("objectName", "field1", ERROR_MESSAGE_1);
+        FieldError fieldError2 = new FieldError("objectName", "field2", ERROR_MESSAGE_2);
+        given(bindingResult.getFieldErrors()).willReturn(asList(fieldError1, fieldError2));
+        given(requestContext.getRequestId()).willReturn(REQUEST_ID);
+
+        // When
+        ResponseEntity<Object> responseEntity = handler.handleBindException(new BindException(bindingResult), httpHeaders, HttpStatus.BAD_REQUEST, webRequest);
+
+        // Then
+        assertThat(responseEntity).isNotNull();
+        assertThat(responseEntity.getBody()).isInstanceOf(ErrorResponse.class);
+        ErrorResponse errorResponse = (ErrorResponse) responseEntity.getBody();
+        assertThat(errorResponse.getRequestId()).isEqualTo(REQUEST_ID);
+        assertThat(errorResponse.getMessage()).isEqualTo(VALIDATION_ERROR_MESSAGE);
+        List<ErrorResponse.FieldError> fieldErrors = errorResponse.getFieldErrors();
+        assertThat(fieldErrors.size()).isEqualTo(2);
+        assertThat(fieldErrors.get(0).getField()).isEqualTo("field1");
```

Comment:
I think these assertions could be refactored to remove duplication by having a method call `assertFieldErrorPresent` that looked for the presence of a FieldError (or field name and message) within the List as ordering isn't really important.

Reply:
Created field error assertion method. Didn't create one using .contains() on the list as that requires additional code to construct the expected fieldError objects.